packages/deploy-tooling/src/base/deploy.ts

```diff
@@ -101,26 +101,45 @@ async function createAbapCloudServiceProvider(
  *
  * @param options - predefined axios options
  * @param target - url target configuration
+ * @param logger - reference to the logger instance
  * @returns {*}  {(Promise<AbapServiceProvider>)}
  */
 async function createAbapServiceProvider(
     options: AxiosRequestConfig,
-    target: UrlAbapTarget
+    target: UrlAbapTarget,
+    logger?: Logger
 ): Promise<AbapServiceProvider> {
+    let abapServiceProvider: AbapServiceProvider | undefined;
     options.baseURL = target.url;
     if (target.client) {
         options.params['sap-client'] = target.client;
     }
     if (!options.auth) {
-        const storedOpts = await getCredentials<BasicAuth>(target);
-        if (storedOpts?.password) {
-            options.auth = {
-                username: storedOpts.username,
-                password: storedOpts.password
-            };
+        const storedOpts = await getCredentials(target);
+        if (storedOpts) {
+            let systemName = '';
+            if ((storedOpts as BasicAuth).password) {
```

Comment:
Could you please create type guard so that you don't need to `as BasicAuth` so often (makes the code unnecessarily complex)